osu.Game/Beatmaps/BeatmapMetadata.cs

```diff
@@ -66,5 +66,23 @@ public string AuthorString
             Source,
             Tags
         }.Where(s => !string.IsNullOrEmpty(s)).ToArray();
+
+        public override bool Equals(object other)
+        {
+            var otherMetadata = other as BeatmapMetadata;
+            if (otherMetadata == null) return false;
+
+            return (onlineBeatmapSetID?.Equals(otherMetadata.onlineBeatmapSetID) ?? false)

```

Comment:
Use `operator ==` for primitive types.
_Defining custom `==` needs some more work, and seems bothering sometimes._

Reply:
Oh right. I will do the same with PreviewTime then.